### ISSUE TITLE
jax.vjp gradients have slow runtimes for the entire forward model

### ISSUE THREAD

**Issue by rgjini** [climate-analytics-lab/jax-gcm#115]
Using jax.vjp to take the gradients of the entire jax-gcm forward model takes very long to run when storing intervals (making save_interval equal to something less than the total model run time).  When the predictions object is empty, the gradients can be computed within a reasonable amount of time, but once the predictions object gets filled, jax.vjp takes a long time to execute.  

Most likely, the issue is with the amount of memory required to take the gradients over the physics data object.  jax.jvp works to calculate parts of the jacobian for individual parameters.  

Items to look into: 

1. Why does jax.jvp seem to take less memory (and less time)? 
2. How can we modify how we are running jax.vjp to make the compute time lower?
